sqllogictest/src/parser.rs

```diff
@@ -293,10 +294,37 @@ impl<T: ColumnType> std::fmt::Display for Record<T> {
     }
 }
 
+/// Whether to enable or disable a feature. Used in `control` statements.
 #[derive(Debug, PartialEq, Eq, Clone)]
+pub enum OnOff {
+    On,
+    Off,
+}
+
+impl OnOff {
+    pub fn try_from_str(s: &str) -> Result<Self, ParseErrorKind> {
+        match s {
+            "on" => Ok(Self::On),
+            "off" => Ok(Self::Off),
+            _ => Err(ParseErrorKind::InvalidControl(s.to_string())),
+        }
+    }
+
+    pub fn as_str(&self) -> &'static str {
+        match self {
+            Self::On => "on",
+            Self::Off => "off",
+        }
+    }
+}
+
+#[derive(Debug, PartialEq, Eq, Clone)]
+#[non_exhaustive]
 pub enum Control {
     /// Control sort mode.
     SortMode(SortMode),
+    /// Control whether or not to substitute variables in the SQL.
+    Substitution(OnOff),
```

Comment:
Why not a simple `bool`? 🤔

Reply:
Okay, then I'll make `try_from_str` and `as_str` a trait.